stac_fastapi/pgstac/tests/api/test_api.py

```diff
@@ -302,3 +316,54 @@ async def test_search_line_string_intersects(
     assert resp.status_code == 200
     resp_json = resp.json()
     assert len(resp_json["features"]) == 1
+
+
+@pytest.mark.asyncio
+async def test_with_default_cors_origin(app_client):
+    resp = await app_client.get("/", headers={"Origin": cors_origin_1})
+    assert resp.status_code == HTTPStatus.OK
+    assert resp.headers["access-control-allow-origin"] == "*"
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("app_client", [{"setup_func": cors_permit_1}], indirect=True)
+async def test_with_match_cors_single(app_client):
+    resp = await app_client.get("/", headers={"Origin": cors_origin_1})
+    assert resp.status_code == HTTPStatus.OK
+    assert resp.headers["access-control-allow-origin"] == cors_origin_1
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("app_client", [{"setup_func": cors_permit_12}], indirect=True)
+async def test_with_match_cors_double(app_client):
+    resp = await app_client.get("/", headers={"Origin": cors_origin_1})
+    assert resp.status_code == HTTPStatus.OK
+    assert resp.headers["access-control-allow-origin"] == cors_origin_1
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "app_client", [{"setup_func": cors_permit_123_regex}], indirect=True
+)
+async def test_with_match_cors_all_regex_match(app_client):
+    resp = await app_client.get("/", headers={"Origin": cors_origin_1})
+    assert resp.status_code == HTTPStatus.OK
+    assert resp.headers["access-control-allow-origin"] == cors_origin_1
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "app_client", [{"setup_func": cors_permit_123_regex}], indirect=True
+)
+async def test_with_match_cors_all_regex_mismatch(app_client):
+    resp = await app_client.get("/", headers={"Origin": cors_origin_deny})
+    assert resp.status_code == HTTPStatus.OK
+    assert "access-control-allow-origin" not in resp.headers
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("app_client", [{"setup_func": cors_deny}], indirect=True)
+async def test_with_mismatch_cors_origin(app_client):
+    resp = await app_client.get("/", headers={"Origin": cors_origin_1})
+    assert resp.status_code == HTTPStatus.OK
+    assert "access-control-allow-origin" not in resp.headers
```

Comment:
Do we need to test the middleware on the stac-fastapi backend? I guess we could just write simple tests using a minimalist fastapi application.

Reply:
I see this as following precedent - I don't see any tests for the `stac_fastapi.api` package, which suggests all "core" functionality is tested exclusively via the backends.

Reply:
can we do like https://github.com/stac-utils/stac-fastapi/blob/master/stac_fastapi/api/tests/test_api.py instead of duplicating those tests

Reply:
@vincentsarago I moved the tests you referenced to a more meaningful filename and extracted to shared code the necessary pieces for my tests to run on api.